### PR TITLE
Add a processing documentation for contributors

### PR DESCRIPTION
**Pull request by jwalgran**
## Overview

Describes how CSVs are processed and facilities are matched in non-technical language.

Connects #268

### Demo

Screen Shot 2019-03-21 at 10 13 30 AM

Screen Shot 2019-03-21 at 10 12 24 AM

Screen Shot 2019-03-21 at 10 12 31 AM

Screen Shot 2019-03-21 at 10 12 52 AM

## Testing Instructions

* Verify that there is an About > Processing option in the navbar and that it links to /about/processing
* Verify the content of /about/processing
* Log in
* Verify that /contribute includes a link to /about/processing
* Verify that the list details page includes a link to /about/processing

### CODE REVIEW

**Comment by kellyi** [comment]
Taking a look!

File: doc/processing.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  - Includes a header row with the following field names (in lower case)
+    - country
+    - name
+    - address
+  - Less than 5MB in size.
+
+Each line in a contributed facility list CSV file is processed separately and
+passes through three steps:
+
+  - _Parse_: The line is separated into distinct country, name, and address fields
```
Comment (kellyi) [inline doc/processing.md]: Consider making these into `###` subheadings, which I believe would make them linkable on github.

File: doc/processing.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+contributor to verify a potential match, the facility will not be searchable
+until the match is approved or rejected. If the application encounters an error
+while processing a row, it will not appear in search results.
+
+> One of the items in my list has an "ERROR" status. How can I tell what happened?
+
+When viewing the list, clicking on a row in an "ERROR" status will show
+additional information regarding why the processing failed.
+
+>How long will it take to process my contributed facility list?
```
Comment (kellyi) [inline doc/processing.md]: Missing space: `>H`

File: doc/processing.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      creates a new facility using the country, name, and address and assigns a new
+      OAR ID number to the new facility.
+
+Contributed rows are matched to the existing facility list using statistical
+text analysis. The application calculates the probability that a contributed row
+is a match to an existing facility by comparing the country, name, and address
+fields individually and then joining those three comparisons together to create
+a final confidence score. The three individual comparisons are not given equal
+weight, however. The country of a submitted row, for example, must be an exact
+match.
```
Comment (kellyi) [inline doc/processing.md]: Would it be too much detail to explain how the name and address scores are weighted?

File: doc/processing.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+The application has a set of examples apparel facility names and addresses that
+are used to "train" the matching process. There are examples of similar items
+that should not be considered matches as well as examples of dissimilar items
+that should be considered matches. Using this training data allows the
+application to mathematically determine which differences in the text of
+facility name or address are significant and which are not.
+
+There are a some categories of text differences that are filtered out before the
```
Comment (kellyi) [inline doc/processing.md]: Typo: "There are a some"

Comment (kellyi) [reply doc/processing.md]: I think this sentence could just begin `Some categories of text differences are filtered out before the matching...`

File: doc/processing.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+>How long will it take to process my contributed facility list?
+
+List items are scheduled for processing as soon as they are uploaded, but it is
+difficult to say with certainty how long it will take to complete the
+processing. The size of the list and the number of other lists being processed
+at the same time will affect how long it will take to complete the processing of
+all the items in a list. If all the items in a contributed list still have the
+initial status of "UPLOADED" after one hour, please send an email to
+info@openapparel.org with the contributor name and the name of the list.
```
Comment (kellyi) [inline doc/processing.md]: Would it be worth adding a question like "How are OAR IDs generated?" whose answer would document how that works?

Comment (kellyi) [reply doc/processing.md]: One more potential question:

```
> What happens when I confirm or reject a match?
```

File: doc/processing.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+processing by uploading lists in comma-separated value (CSV) format. All popular
+spreadsheet programs, including Microsoft Excel, can save a sheet in CSV format.
+The Open Apparel Registry provides a downloadable template, but any CSV can be
+read and processed if it meets the following criteria:
+
+  - Includes a header row with the following field names (in lower case)
+    - country
+    - name
+    - address
+  - Less than 5MB in size.
```
Comment (kellyi) [inline doc/processing.md]: May be worth enumerating another criterion here which is something like

```
- Has facility rows made up of comma separated country, name, and address fields
```

File: doc/processing.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+Each line in a contributed facility list CSV file is processed separately and
+passes through three steps:
+
+  - _Parse_: The line is separated into distinct country, name, and address fields
+    and the country field is verified against a list of known countries.
+
+  - _Geocode_: The address and country are submitted to a service that returns a
```
Comment (jwalgran) [inline doc/processing.md]: location _on_ the globe

File: doc/processing.md (hunk 3 above)
Comment (jwalgran) [reply doc/processing.md]: I think getting into the details of the statistics is out of scope.

File: doc/processing.md (hunk 7 above)
Comment (jwalgran) [reply doc/processing.md]: Fixed in 4d2c0ab

File: doc/processing.md (hunk 6 above)
Comment (jwalgran) [reply doc/processing.md]: Fixed in 4d2c0ab

File: doc/processing.md (hunk 5 above)
Comment (jwalgran) [reply doc/processing.md]: 👍 Natalie and Katie liked these questions. I will add them.

Comment (jwalgran) [reply doc/processing.md]: Added in 0a0426d

File: doc/processing.md (hunk 4 above)
Comment (jwalgran) [reply doc/processing.md]: Fixed in 4d2c0ab

File: doc/processing.md (hunk 1 above)
Comment (jwalgran) [reply doc/processing.md]: Fixed in 4d2c0ab

**Comment by jwalgran** [comment]
I have incorporated the PR comments and feedback from Katie and Natalie pulled from https://docs.google.com/document/d/1oHidZH3h55pvtQOP93ewU1VFtOKh13hpcYHdHo42IrY

Please review the responses to the new questions. Thanks.

File: doc/processing.md (hunk 6 above)
Comment (kellyi) [reply doc/processing.md]: We may have decided to drop this change, but I don't see it in the updated version either.

**Comment by jwalgran** [comment]
I componentized this and linked to it from the Navbar, Contribute and FacilityListItems components.

Please give this another look.

**Comment by jwalgran** [comment]
I have updated the PR description

**Comment by kellyi** [comment]
Looking good!

Screen Shot 2019-03-21 at 1 17 42 PM

**Pull request merged**